GUIDE_advanced_manual.md

```diff
@@ -0,0 +1,141 @@
+
+- Create a copy of the shader that is used by the materials of your avatar.
+- Add a property to enable the fake arm.
+- **For every shader pass:**
+    - Add vertex color information for use by the geometry shader.
+    - Add a geometry shader or edit the existing one.
+    - In the geometry shader, return immediately if the fake arm is enabled AND the values of green and blue are zero.
+
+## Text editor: Create a shader to show the fake arm
+
+The file `ExampleUnlitHandholdingShaderIK/UnlitHandholdingShaderIK.shader` contains the minimal changes that have been applied to an Unlit shader. Edit your main shader to integrate the same changes, which summarizes to:
+
+- Create a copy of the shader that is used by the materials of your avatar.
+- **Only keep the shader pass that has `"Lightmode"="ForwardBase"` in the Tags. Delete all the other shader passes.**
+- Add a property to enable the fake arm.
+- Add a property to select the black light intensity value.
+- Add other properties if you need them.
+- Expose vertex color information to the vertex shader.
+- Find a good insertion point in the existing the vertex shader.
+- Scale the input vertex by 100 in order to make it visible to users who have shaders enabled.
+- Include `HaiHandholdingShaderIK/HaiHandholdingShaderIK.cginc`
+- Call the `transformArm` function (see example shader).
+- Pass the resulting value of that function call to the rest of the shader.
+- Add a geometry shader or edit the existing one.
+- In the geometry shader, return immediately if the fake arm is disabled.
+
+## Blender: Mark the arm of main avatar
+
+In a copy of your main avatar:
+
+- Select the vertex groups of hand & fingers, forearm and upper arm. Don't select the shoulder.
+- Use Vertex painting to set these vertices RED (1.0, 0.0, 0.0).
+- Save the model.
+- Export this model to FBX.
+
+## Blender: Create a fake arm
+
+Create another copy of your main avatar:
+
+- Pose the fingers of the hand to be just a little cramped instead of flat. You can use CATS plugin for that using Start pose mode/Apply as rest pose.
+
+- In Edit mode:
+    - Select the vertex groups of hand & fingers, forearm, upper arm, AND the shoulder.
+    - Invert the selection.
+    - Delete the faces.
+    - Rotate the hand so that the orientation is different, as if you were grabbing a vertical pole in the subway or in the bus.
+      - I recommend selecting the vertices of the hand and perform a proportional editing to twist the forearm and a very small part of the upper arm.
+      - Alternatively, you may use Pose mode for that.
+    - Select the vertex group of hand & fingers and forearm.
+      - Set the vertex color to RED (1.0, 0.0, 0.0).
+    - Select the vertex group of upper arm.
+      - Set the vertex color to GREEN (0.0, 1.0, 0.0).
+    - Select all vertices.
+      - Perform a vertex paint smoothing.
```

Comment:
May need to find another way to copy the vertex paint. Mandate usage of another lightweight editor script here?